### PR TITLE
Support `account_nextIndex` RPC.

### PR DESCRIPTION
**Pull request by tomusdrw**
Exposes `account_nextIndex` RPC for polkadot.

I'm not super happy with this PR, since we use `node-*` packages (especially `node-primitives`), but I'm not quite sure yet how to deal with it.

1. One idea would be to move `AccountsNonceApi` (RuntimeApi) to `srml::system` (but then we have strict dependencies on particular `srml`s`.
2. Second I had was to actually automate registering this kind of `RuntimeApi`-based RPCs so that we don't have to write this glue code.
3. I'm a bit hesitant on putting the runtime API and the RPC implementation to `core` as it's very specific for a particular `srml` set. Is there a place that's kind of between `core` and `srml` (i.e. depends on both, but is not `node`)? Should we introduce one?

## Update

Updated the PR to support latest `master` now. Still keeping `onice` label until the refactoring is merged to `polkadot-master` branch.

### CODE REVIEW

**Comment by gavofyork** [comment]
Ok - I think it's fine to have this additional off-chain logic that specifically relates to an SRML module in the `srml` path, probably in its own crate in a sub path of `srml/system`.

**Comment by rphmeier** [comment]
Right - it's also important to keep the separations clear so that we could have separate licenses for core, srml, node, polkadot, etc.

**Comment by tomusdrw** [comment]
I'm working on a PR to substrate currently to move this stuff to the right places, put this one on hold until it's merged there.

File: rpc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+[dependencies]
+client = { package = "substrate-client", git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
+jsonrpc-core = "13.2.0"
+polkadot-primitives = { path = "../primitives" }
+sr-primitives = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master"  }
+substrate-rpc = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master" }
+transaction_pool = { package = "substrate-transaction-pool", git = "https://github.com/paritytech/substrate", branch = "polkadot-master"  }
+srml-system-rpc = { git = "https://github.com/paritytech/substrate", branch = "polkadot-master"  }
+
```
Comment (bkchr) [inline rpc/Cargo.toml]:
```suggestion
```

File: service/src/lib.rs [hunk 2]
```diff
@@ -117,7 +117,11 @@ macro_rules! new_full_start {
 
 				import_setup = Some((block_import, grandpa_link, babe_link));
 				Ok(import_queue)
-			})?;
+			})?
+			.with_rpc_extensions(|client, pool| -> polkadot_rpc::RpcExtension {
+				polkadot_rpc::create(client, pool)
+			})?
```
Comment (bkchr) [inline service/src/lib.rs]:
```suggestion
			})?;
```

File: service/src/lib.rs [hunk 3]
```diff
@@ -117,7 +117,11 @@ macro_rules! new_full_start {
...
+			;
```
Comment (bkchr) [inline service/src/lib.rs]:
```suggestion
```

**Pull request merged**